src/script/descriptor.cpp

```diff
@@ -1418,6 +1420,59 @@ class RawTRDescriptor final : public DescriptorImpl
     }
 };
 
+/** A parsed rawnode(...) descriptor */
+class RawNodeDescriptor final : public DescriptorImpl
+{
+    std::vector<unsigned char> m_bytes;
+protected:
+    std::string ToStringExtra() const override { return HexStr(m_bytes); }
+    std::vector<CScript> MakeScripts(const std::vector<CPubKey>&, Span<const CScript>, FlatSigningProvider&) const override { return Vector(CScript(m_bytes.begin(), m_bytes.end())); }
+public:
+    RawNodeDescriptor(std::vector<unsigned char> bytes) : DescriptorImpl({}, "rawnode"), m_bytes(std::move(bytes)) {}
+
+    bool IsSolvable() const final { return false; }
+
+    bool IsSingleType() const final { return true; }
+    bool ToPrivateString(const SigningProvider& arg, std::string& out) const final { return false; }
+
+    std::optional<int64_t> ScriptSize() const override { return m_bytes.size(); }
+
+    std::unique_ptr<DescriptorImpl> Clone() const override
+    {
+        return std::make_unique<RawNodeDescriptor>(m_bytes);
+    }
+};
+
+/** A parsed rawleaf(...) descriptor */
+class RawLeafDescriptor final : public DescriptorImpl
+{
+    CScript m_leaf_script;
+    uint8_t m_leaf_version;
+protected:
+    std::string ToStringExtra() const override {
+        return strprintf("%s,%x", HexStr(m_leaf_script), m_leaf_version);
+    }
+    std::vector<CScript> MakeScripts(const std::vector<CPubKey>&, Span<const CScript>, FlatSigningProvider&) const override { return Vector(m_leaf_script); }
+public:
+    RawLeafDescriptor(CScript leaf_script, uint8_t leaf_version) : DescriptorImpl({}, "rawleaf"), m_leaf_script(leaf_script), m_leaf_version(leaf_version) {}
+
+    bool IsSolvable() const final { return false; }
```

Comment:
In  "descriptor: Implement rawnode() partial descriptor" a9a9e78c3f839eceebb3abd473d8658fabd92f1f
We are duplicating this code here!

I wonder if `DescriptorImpl` will just define this duplicate methods defined with the truthy or falsy values,  All descriptor that require concrete implementation will just override those methods.